packages/coil-extension/src/content/services/RunContentHandler.ts

```diff
@@ -28,18 +34,43 @@ export class RunContentHandler {
     private client: GraphQlClient
   ) {}
 
-  async adaptedPageDetails(url: string) {
+  async adaptedPageDetails(url: string, site: string) {
     debug('fetching payment pointer for this page', url)
 
+    // Wait for page to load to ensure there's a channel url to grab
+    if (document.readyState !== 'complete' && site === 'youtube') {
```

Comment:
Try and use document/window via DI so as this gets restructured it's in a good shape for tests with mocking